test/jdk/javax/net/ssl/DTLS/DTLSOverDatagram.java

```diff
@@ -581,58 +577,55 @@ public final void runTest(DTLSOverDatagram testCase) throws Exception {
         }
     }
 
-    record ServerCallable(DTLSOverDatagram testCase, DatagramSocket socket,
-                          InetSocketAddress clientSocketAddr) implements Callable<String> {
-
-        @Override
-        public String call() throws Exception {
-            try {
-                testCase.doServerSide(socket, clientSocketAddr);
-            } catch (Exception e) {
-                System.out.println("Exception in  ServerCallable.call():");
-                e.printStackTrace(System.out);
-                serverException = e;
-
-                if (testCase.isGoodJob()) {
-                    throw e;
-                } else {
-                    return "Well done, server!";
-                }
-            }
+    String runServer(DTLSOverDatagram testCase, DatagramSocket socket,
+                          InetSocketAddress clientSocketAddr) throws Exception {
+        try {
+            serverStarted.countDown();
+            testCase.doServerSide(socket, clientSocketAddr);
+        } catch (Exception e) {
+            System.out.println("Exception in  ServerCallable.call():");
+            e.printStackTrace(System.out);
+            serverException = e;
 
             if (testCase.isGoodJob()) {
-                return "Well done, server!";
+                throw e;
             } else {
-                throw new Exception("No expected exception");
+                return "Well done, server!";
             }
         }
-    }
 
-    record ClientCallable(DTLSOverDatagram testCase, DatagramSocket socket,
-                          InetSocketAddress serverSocketAddr) implements Callable<String> {
+        if (testCase.isGoodJob()) {
+            return "Well done, server!";
+        } else {
+            throw new Exception("No expected exception");
+        }
+    }
 
-        @Override
-        public String call() throws Exception {
-            try {
+    private String runClient(DTLSOverDatagram testCase, DatagramSocket socket, InetSocketAddress serverSocketAddr)
+    throws Exception {
+        try {
+            if(!serverStarted.await(5, TimeUnit.SECONDS)) {
+                throw new Exception("Server did not start within 5 seconds.");
+            } else {
                 testCase.doClientSide(socket, serverSocketAddr);
-            } catch (Exception e) {
-                System.out.println("Exception in ClientCallable.call():");
-                e.printStackTrace(System.out);
-                clientException = e;
-
-                if (testCase.isGoodJob()) {
-                    throw e;
-                } else {
-                    return "Well done, client!";
-                }
             }
+        } catch (Exception e) {
+            System.out.println("Exception in ClientCallable.call():");
+            e.printStackTrace(System.out);
+            clientException = e;
 
             if (testCase.isGoodJob()) {
```

Comment:
Same here. Check if this is method is even used and what is the purpose of it. You may not need these checks and would prefer to have no try-catch in this method.

Reply:
It is used by InvalidRecords.java test. It expects handshake to fail so overrides this method.

Reply:
refactored the code to be a bit cleaner.

The InvalidRecords.java test explicitly threw an exception that is then treated as a successful test run. I refactored the test a little so it only throws an exception if the test fails.